File: Ahorn/triggers/lightningStrikeTrigger.jl
module YetAnotherHelperLightningStrikeTrigger

using ..Ahorn, Maple

@mapdef Trigger "YetAnotherHelper/LightningStrikeTrigger" LightningStrikeTrigger(x::Integer, y::Integer, width::Integer=Maple.defaultTriggerWidth, height::Integer=Maple.defaultTriggerHeight, playerOffset::Number=0.0, verticalOffset::Number=0.0, strikeHeight::Number=0.0, seed::Integer=0, delay::Number=0.0, rain::Bool=true, flash::Bool=true, constant::Bool=false)

const placements = Ahorn.PlacementDict(
    "Lightning Strike (Yet Another Helper)" => Ahorn.EntityPlacement(
        LightningStrikeTrigger,
        "rectangle",
    ),
)

end

File: Loenn/triggers/lightningStrikeTrigger.lua
local lightningStrikeTrigger = {}

lightningStrikeTrigger.name        = "YetAnotherHelper/LightningStrikeTrigger"
lightningStrikeTrigger.placements  = {
    name = "normal",
    data = {
        playerOffset    = 0.0,
        verticalOffset  = 0.0,
        strikeHeight    = 0.0,
        seed            = 0,
        delay           = 0.0,
        raining         = true,
        flash           = true,
        constant        = false
    }
}

return lightningStrikeTrigger
